### PR TITLE
chore: bump Storybook deps

### PR DESCRIPTION
**Pull request by nschonni**
Closes #3396
Closes #3393

### CODE REVIEW

**Comment by nschonni** [comment]
@peterbe the pinning seems kind of gross, but it gets around the Dependabot PR issue

**Comment by peterbe** [comment]
Sorry. It seems Dependabot managed to sneak in and make your files conflicting.

File: package.json [hunk 1]
```diff
@@ -137,6 +139,7 @@
     "react": "17.0.2",
     "react-dom": "17.0.2",
     "react-is": "^17.0.2",
+    "react-refresh": "0.9",
```
Comment (peterbe) [inline package.json:142]: Isn't that already in place thanks to `react-scripts`?
I know we've probably discussed this many times but won't having it here just cause another problem later when `react-scripts` decides it needs `react-refresh@1.0`?

Comment (nschonni) [reply package.json:142]: It already complains if it doesn't have the version it likes 😆 , that's why I had to pin the `babel-loader` version that Storybook wanted, but it didn't like

**Comment by peterbe** [comment]
So sorry @nschonni I have no excuse except the fact that I was on vacation all of last week. 
Would you mind trying to merge in `main` again. 

Would be nice @schalkneethling could take a look too.

**Comment by nschonni** [comment]
> So sorry @nschonni I have no excuse except the fact that I was on vacation all of last week.

No excuse necessary 😄

**Review by peterbe — APPROVED** [review]

**Pull request merged**